pennylane/ops/op_math/parametrized_hamiltonian.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This submodule contains the ParametrizedHamiltonian class
+"""
+import pennylane as qml
+from pennylane.operation import Operator
+from pennylane.ops.qubit.hamiltonian import Hamiltonian
+from pennylane.wires import Wires
+
+
+# pylint: disable= too-many-instance-attributes
+class ParametrizedHamiltonian:
+    r"""Callable object holding the information representing a parametrized Hamiltonian.
+
+    Passing parameters to the ``ParametrizedHamiltonian`` returns an :class:`~pennylane.operation.Operator` representing the Hamiltonian
+    for that set of parameters.
+
+    The Hamiltonian can be represented as a linear combination of other operators, e.g.,
+    :math:`H(v, t) = H_\text{drift} + \sum_j f_j(v, t) H_j`, where the :math:`v` are trainable parameters,
+    and t is time.
+
+    For example, a time-dependent ``ParametrizedHamiltonian`` with a single trainable parameter, :math:`a`, could be :math:`H = 2 X_1 X_2 + \sin(a t) Y_1 Y_2`
+
+    Args:
+        coeffs (Union[float, callable]): coefficients of the Hamiltonian expression, which may be constants or
+            parametrized functions. All functions passed as ``coeffs`` must have the signature ``(params, t)``
+            with ``params`` being trainable parameters and ``t`` being time.
+        observables (Iterable[Observable]): observables in the Hamiltonian expression, of same length as ``coeffs``
+
+    A ``ParametrizedHamiltonian`` is callable with the fixed signature ``H(params, t)`` that returns an
+    ``Operator`` representing an instance of the Hamiltonian with the specified parameter values.
+
+    **Example:**
+
+    .. code-block:: python3
+
+        coeffs = [2, lambda v, t: v[0] * jnp.sin(v[1] * t)]
+        observables =  [qml.PauliX(0), qml.PauliY(1)]
+        H = ParametrizedHamiltonian(coeffs, observables)
+
+    >>> H(jnp.ones(2), 1.)
+    (2*(PauliX(wires=[0]))) + (0.8414710164070129*(PauliY(wires=[1])))
```

Comment:
awesome! Minor suggestion to add already the `params` behavior (optional, your call)
```suggestion
    .. code-block:: python3

        coeffs = [2., lambda v, t: v[0] * jnp.sin(v[1] * t), lambda v, t: jnp.polyval(params[2:8], t)]
        observables =  [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
        H = ParametrizedHamiltonian(coeffs, observables)

    >>> v = jnp.ones(8, dtype=float)
    >>> H(v, 1.)
    (2*(PauliX(wires=[0]))) + (0.8414710164070129*(PauliY(wires=[1]))) # need to modify the output
```
(note that you need to modify the output)

Reply:
This is not needed due to the UI change. Marking as resolved.